org.eclipse.scanning.connector.epicsv3/src/org/eclipse/scanning/connector/epicsv3/EpicsV3DynamicDatasetConnector.java

```diff
@@ -277,9 +249,12 @@ public String connect(long time, TimeUnit unit) throws DatasetException {
 
 	@Override
 	public void disconnect() throws DatasetException {
-		ec.destroy(dataChannel);
-		ec.destroy(dim0Ch);
-		ec.destroy(dim1Ch);
+		if (dataChannel != null)	ec.destroy(dataChannel);
+		if (dim0Ch != null)			ec.destroy(dim0Ch);
+		if (dim1Ch != null)			ec.destroy(dim1Ch);
+		if (dim2Ch != null)			ec.destroy(dim2Ch);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') At most one statement is allowed per line, but 2 statements were found on this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00122)